nwbwidgets/timeseries.py

```diff
@@ -966,4 +942,485 @@ def update(
         fig = fig if fig is not None else go.FigureWidget()
         fig.data = []
         fig.layout = {}
-        return self.plot_group(group_inds,data,time_ts_aligned,fig,order)
+        return self.plot_group(group_inds, data, time_ts_aligned, fig, order)
+
+
+def get_time_series_data(
+    time_series: TimeSeries,
+    idx_start: Optional[int] = None,
+    idx_stop: Optional[int] = None,
+    data_column: Optional[int] = None,
+    scale_data: bool = True,
+):
+
+    if (data_column is not None) and time_series.data.ndim > 1:
+        data = time_series.data[idx_start:idx_stop, data_column]
+    else:
+        data = time_series.data[idx_start:idx_stop]
+
+    # Scale data
+    if scale_data:
+        offset_scalar = time_series.offset
+        conversion_factor_scalar = time_series.conversion
+        channel_conversion_vector = (
+            time_series.channel_conversion if hasattr(time_series, "channel_conversion") else np.ones_like(data)
+        )
+
+        data = data * channel_conversion_vector * conversion_factor_scalar + offset_scalar
+    return data.flatten()
+
+
+def trialize_time_series(
```

Comment:
Could you use this function?

https://github.com/NeurodataWithoutBorders/nwbwidgets/blob/dba14008af7f7aea6f3ea83f010a0a89cca1bfa4/nwbwidgets/utils/timeseries.py#L266

Reply:
Most of the function implemented there is to build the dataframe into the form where the faceting functionality of plotly can be used.

There is a place inside `trialized_time_series` where the data from trials is extracted and also the timestamps. For that part, after your last suggestion above, I took a look a the functions in that utils module for timeseries. I don't think they fit that well here without introducing more unnecessary computation. 

Basically, right now I have a vectorized bisect function (as provided by numpy) which I use to extract the set of trialized indexes and then I use them on both the data and the timestamps. The functions in the utils module are non vectorized and they return the data in a list without the indexes which I need for the timestamps. Using `align_by_time_intervals`, for example, would reduce the performance here both by losing the vectorization and then by requiring some more computation to get the indexes of the timestamps.

I think the generalization would be to have a general vectorized function that return the trialized indexes. That hypothetical function then can be used both here and inside `bisect_timeseries_by_times` to avoid the per trial bisection there. 

What do you think?